### PR TITLE
[MAINTENANCE] Make unfiltered warnings into errors

### PR DESCRIPTION
**Pull request by anthonyburdi**
To make sure unhandled warnings don't show when users are running our agent, this change will cause CI to fail in a PR if a warning is raised during any test.

### CODE REVIEW

**Comment by Kilo59** [comment]
Most of the warnings seem to be coming from `great_expectations` and should be filtered/handled there if the goal is to keep our users from seeing them.

**Review by Kilo59 — APPROVED** [review]

File: poetry.lock [hunk 1]
```diff
@@ -796,7 +796,6 @@ requests = ">=2.20"
 "ruamel.yaml" = ">=0.16,<0.17.18"
 scipy = ">=1.6.0"
 sqlalchemy = {version = ">=1.4.0,<2.0.0", optional = true, markers = "extra == \"databricks\" and extra == \"redshift\""}
-sqlalchemy-redshift = {version = ">=0.8.8", optional = true, markers = "extra == \"redshift\""}
```
Comment (Shinnnyshinshin) [inline poetry.lock:799]: Great Expectations installed without `sqlalchemy-redshift`

File: poetry.lock [hunk 2]
```diff
@@ -3530,21 +3529,6 @@ postgresql-psycopg2cffi = ["psycopg2cffi"]
...
-python-versions = ">=3.4"
-files = [
-    {file = "sqlalchemy-redshift-0.8.14.tar.gz", hash = "sha256:1f1f78d8ef7febaf0553d64fe247fc08acbfec84d8e1d2dc8264c656cc623622"},
-    {file = "sqlalchemy_redshift-0.8.14-py2.py3-none-any.whl", hash = "sha256:cfdfae2c8fb3043c181e2baedd4ee425c1ca7efed20f3c5ae274838d1015f919"},
-]
-
-[package.dependencies]
-packaging = "*"
-SQLAlchemy = ">=0.9.2,<2.0.0"
-
```
Comment (Shinnnyshinshin) [inline poetry.lock:3547]: `sqlalchemy-redshift` not installed in poetry env.

File: poetry.lock [hunk 3]
```diff
@@ -3533,9 +3532,9 @@ sqlcipher = ["sqlcipher3-binary"]
 [[package]]
 name = "sqlalchemy-redshift"
 version = "0.8.14"
-description = "Amazon Redshift Dialect for sqlalchemy"
+description = "Amazon Redshift Dialect for sqlalchemy - Currently waiting for the next version 0.8.15 to be released where PR https://github.com/sqlalchemy-redshift/sqlalchemy-redshift/pull/288 will have been merged."
 optional = false
-python-versions = ">=3.4"
+python-versions = "<3.6"
```
Comment (Shinnnyshinshin) [inline poetry.lock:3537]: Added longer description and python version where package would work

File: poetry.lock (hunk 1 above)
Comment (Kilo59) [reply poetry.lock:799]: This is not about file you should manually modifying

**Review by Kilo59 — CHANGES_REQUESTED** [review]
You should not be manually modifying the lock file

File: poetry.lock (hunk 1 above)
Comment (Kilo59) [reply poetry.lock:799]: You need to modify the poetry dependencies specified in pyproject.toml and then poetry can regenerate a new lock file.

Comment (Kilo59) [reply poetry.lock:799]: This likely also means removing `redshift` from the `gx extras` and specifying it separately. 
```toml
 [tool.poetry.group.sql.dependencies] 
 great-expectations = { version = "^0.18.1", extras = [ 
     "databricks", 
     "sqlalchemy1", 
     "postgres", 
 ] } 
 snowflake-sqlalchemy = ">=1.5.0" 
 snowflake-connector-python = ">=3.3.1" 
# TODO: verify the actual versions here
whatever-this-redshift-dep-is = {version: "^1.0", python: "<3.11"}
```

File: poetry.lock
Comment (Kilo59) [inline poetry.lock:1]: You need to revert these changes and use `poetry` to regenerate the lock file.https://python-poetry.org/docs/cli/#lock

File: pyproject.toml [hunk 4]
```diff
@@ -35,11 +35,10 @@ great-expectations = { version = "^0.18.1", extras = [
     "databricks",
     "sqlalchemy1",
     "postgres",
-    "redshift",
 ] }
 snowflake-sqlalchemy = ">=1.5.0"
 snowflake-connector-python = ">=3.3.1"
-
+sqlalchemy-redshift = {version = "^0.8.8", python = "<3.6"}
 [tool.poetry.group.dev.dependencies]
```
Comment (Shinnnyshinshin) [inline pyproject.toml:42]: 🚀  Thank you for the help here @Kilo59

[the remark this answers is not in the document]
Comment (Shinnnyshinshin) [reply pyproject.toml]: Nope it's definitely all of them @Kilo59. 

According to the [redshift-sqlalchemy](https://github.com/sqlalchemy-redshift/sqlalchemy-redshift/pull/288) PR, the escape character errors are a result of a change in 3.6

Screenshot 2023-12-07 at 2 23 27 PM

[the remark this answers is not in the document]
Comment (Kilo59) [reply pyproject.toml]: These should be causing deprecation warnings in earlier versions and only run time failures in the latest python.

Are you saying Redshift doesn't work at all in Python > 3.6?

[the remark this answers is not in the document]
Comment (Shinnnyshinshin) [reply pyproject.toml]: good question @Kilo59. that's what i'm trying to wrap my head around too. I'll dig into this right now.

File: pyproject.toml [hunk 5]
```diff
@@ -173,4 +173,32 @@ markers =[
     "agentjobs",
     "unit",
 ]
-log_level = "info"
+log_level = "error"
```
Comment (Kilo59) [inline pyproject.toml:176]:
```suggestion
log_level = "info"
```

**Review by Kilo59 — APPROVED** [review]
Looks good.
You might add a comment and todo above `sqlalchemy-redshift` in `pyproject.toml` to remove the python constraint once this https://github.com/sqlalchemy-redshift/sqlalchemy-redshift/pull/288 gets merged.

File: pyproject.toml [hunk 6]
```diff
@@ -174,3 +175,31 @@ markers =[
     "unit",
 ]
 log_level = "info"
+filterwarnings = [
+    #"error",  # Turn all warnings not explicitly filtered below into errors
```
Comment (Kilo59) [inline pyproject.toml:179]: @Shinnnyshinshin you forgot to uncomment this line.
When I do I still see the same errors.
